torch/csrc/dynamo/guards.cpp

```diff
@@ -886,6 +886,11 @@ std::string get_exception_message() {
 }
 
 bool is_immutable_object(py::handle example_value) {
+  static py::object config_module = py::module_::import("torch._dynamo.config");
+  bool is_tensor_immutable =
+      config_module.attr("skip_tensor_guards_with_matching_dict_tags")
+          .cast<bool>();
```

Comment:
Doing this here seems suboptimal:
1) It is run *every* time this function is called, even if called with a non-tensor type (e.g. long/float/etc) -- this seems slow
2) Should we be doing this check when the guards are created rather than when they are evaluated?  Most other dynamo configs are evaluated at compile time.

Reply:
This function is run at the construction time. The return of this function is stored in the guard manager state.